src/Microsoft.SqlTools.ServiceLayer/Connection/SqlConnectionFactory.cs

```diff
@@ -24,5 +26,36 @@ public DbConnection CreateSqlConnection(string connectionString, string azureAcc
             RetryPolicy commandRetryPolicy = RetryPolicyFactory.CreateDefaultConnectionRetryPolicy();
             return new ReliableSqlConnection(connectionString, connectionRetryPolicy, commandRetryPolicy, azureAccountToken);
         }
+        /// <summary>
+        /// Safely casts DbConnection into SqlConnection
+        /// </summary>
+        public SqlConnection ConvertDBConnectionToSqlConnection(DbConnection connection)
+        {
+            // Get a connection to the database for SMO purposes
+            SqlConnection sqlConn = connection as SqlConnection;
+            if (sqlConn == null)
+            {
+                // It's not actually a SqlConnection, so let's try a reliable SQL connection
+                ReliableSqlConnection reliableConn = connection as ReliableSqlConnection;
+                if (reliableConn == null)
+                {
+                    // If we don't have connection we can use with SMO, just give up on using SMO
+                    return null;
+                }
+
+                // We have a reliable connection, use the underlying connection
+                sqlConn = reliableConn.GetUnderlyingConnection();
+            }
+            return sqlConn;
+        }
+
+        /// <summary>
+        /// Check if authorization is for a contained database user.
+        /// </summary>
+        public bool CheckIfContainedAuthorization(DbConnection connection)
+        {
+            SqlConnection sqlConnection = this.ConvertDBConnectionToSqlConnection(connection);
+            return new ServerConnection(sqlConnection).IsContainedAuthentication;
```

Comment:
should `sqlConnection` be null  checked before being used to create a ServerConnection?